astropy/time/formats.py

```diff
@@ -691,6 +691,156 @@ def to_value(self, timezone=None, parent=None):
+        dict of ndarrays that can be used easily by ``set_jds()``.  This is useful
+        because it makes it easy to get default values in that routine.
+
+        Parameters
+        ----------
+        val1 : ndarray or None
+        val2 : ndarray or None
+
+        Returns
+        -------
+        val1_as_dict, val2 : val1 as dict or None, val2 is always None
+
+        """
+        if val2 is not None:
+            raise ValueError('val2 must be None for ymdhms format')
+
+        ymdhms = ['year', 'month', 'day', 'hour', 'minute', 'second']
+
+        if val1.dtype.names:
+            # NumPy structured array or row thereof
+            if not set(val1.dtype.names).issubset(set(ymdhms)):
+                raise ValueError(f'input column names {val1.dtype.names}'
+                                 f'must be subset of {ymdhms}')
+            # Convert to a dict of ndarray
+            val1_as_dict = {name: val1[name] for name in val1.dtype.names}
+
+        elif val1.shape == (0,):
+            # Input was empty list [], so set to None and set_jds will handle this
+            val1_as_dict = None
+
+        elif val1.dtype.kind == 'O':
+            # Code gets here for input as a dict or list of dict.  The dict input
+            # can be either scalar values or N-d arrays.  If the input was some
+            # other inappropriate object (giving dtype=object) then the first if-block
+            # will raise a ValueError.
+
+            # Turn it back into original object (a dict) or list of objects
+            val1_as_dict = val1.tolist()
+
+            if not isinstance(val1_as_dict, dict):
+                # Try converting list of dict to dict of list.  If this fails for any
+                # reason then the input does not meet API requirements for this format.
+                out = collections.defaultdict(list)
+                try:
+                    # If input is an iterable of dict then this will work, otherwise
+                    # the input was not compatible with YMDHMS.
+                    for row in val1_as_dict:
+                        for key, value in row.items():
+                            out[key].append(value)
+                except Exception:
+                    raise ValueError(f'input must be dict, list of dict, or table-like')
+
+                val1_as_dict = out
+
+            # Ensure the dict keys are good
+            if not set(val1_as_dict).issubset(set(ymdhms)):
+                raise ValueError(f'input key names {val1_as_dict.keys()}'
+                                 f'must be subset of {ymdhms}')
+
+            # Put this back into val1 and make all dict values into ndarray
+            for key, value in val1_as_dict.items():
+                val1_as_dict[key] = np.asarray(value)
+
+            # Check that all dict value shapes are the same
+            shapes = set([value.shape for value in val1_as_dict.values()])
```

Comment:
This seems unnecessarily restrictive: it would seem fine to allow broadcastable shapes.